src/main/java/com/firebolt/jdbc/resultset/FireboltResultSet.java

```diff
@@ -562,45 +564,35 @@ public <T> T unwrap(Class<T> iface) throws SQLException {
 	}
 
 	@Override
-	@NotImplemented
-	@ExcludeFromJacocoGeneratedReport
 	public InputStream getAsciiStream(int columnIndex) throws SQLException {
-		throw new FireboltUnsupportedOperationException();
+		return getBinaryStream(columnIndex);
 	}
 
 	@Override
 	@NotImplemented
 	@ExcludeFromJacocoGeneratedReport
 	public InputStream getUnicodeStream(int columnIndex) throws SQLException {
-		throw new FireboltSQLFeatureNotSupportedException();
+		return getBinaryStream(columnIndex);
 	}
 
 	@Override
-	@NotImplemented
-	@ExcludeFromJacocoGeneratedReport
 	public InputStream getBinaryStream(int columnIndex) throws SQLException {
-		throw new FireboltUnsupportedOperationException();
+		return ofNullable(getString(columnIndex)).map(String::getBytes).map(ByteArrayInputStream::new).orElse(null);
 	}
 
 	@Override
-	@NotImplemented
-	@ExcludeFromJacocoGeneratedReport
 	public InputStream getAsciiStream(String columnLabel) throws SQLException {
-		throw new FireboltUnsupportedOperationException();
+		return getBinaryStream(columnLabel);
 	}
 
 	@Override
-	@NotImplemented
-	@ExcludeFromJacocoGeneratedReport
 	public InputStream getUnicodeStream(String columnLabel) throws SQLException {
-		throw new FireboltSQLFeatureNotSupportedException();
+		return getBinaryStream(columnLabel);
 	}
```

Comment:
In these methods Postgres ensures only ASCII/Unicode values are returned. I think we should do the same, otherwise the user might get a unicode value from `getAsciiStream`, which is just incorrect. https://github.com/pgjdbc/pgjdbc/blob/aa235e14c2213d1386100e62206006d83082d914/pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java#L2859-L2894

Reply:
Good catch. Done